src/Spago/Build.hs

```diff
@@ -87,10 +89,26 @@ build BuildOptions{..} maybePostBuild = do
 repl :: Spago m => [Purs.SourcePath] -> [Purs.ExtraArg] -> m ()
 repl sourcePaths passthroughArgs = do
   echoDebug "Running `spago repl`"
-  config <- Config.ensureConfig
-  deps <- Packages.getProjectDeps config
-  let globs = Packages.getGlobs deps <> Config.configSourcePaths config <> sourcePaths
-  Purs.repl globs passthroughArgs
+
+  hasPackagesDhall <- testfile "packages.dhall"
+  if hasPackagesDhall
+    then do
+      config <- Config.ensureConfig
+      deps <- Packages.getProjectDeps config
+      let globs = Packages.getGlobs deps <> Config.configSourcePaths config <> sourcePaths
+      Purs.repl globs passthroughArgs
+    else Temp.withSystemTempDirectory "spago-repl-tmp" $ \dir -> do
+      Turtle.cd (Turtle.decodeString dir)
+
+      Packages.initProject False
+
+      config <- Config.parseConfig
+      deps <- Packages.getProjectDeps config
+      let globs = Packages.getGlobs deps <> Config.configSourcePaths config <> sourcePaths
+
+      Packages.install Nothing Nothing (fmap fst deps)
```

Comment:
Should the flags to `install` come from the user?

Reply:
Yes! Also note that we don't need the full blown `install`, but we can call just the `fetchPackages` part here, e.g. see
https://github.com/spacchetti/spago/blob/92e9c64bdde0373a64514def96b7f78738601909/src/Spago/Packages.hs#L169